src/app/file-manager/transferables-grid/transferables-grid.component.ts

```diff
@@ -191,4 +204,12 @@ export class TransferablesGridComponent implements OnInit, AfterViewInit {
 
   sortData(evt) {
   }
+
+  startDownload() {
+    /* Info needed to download file:
+    console.log(JSON.stringify(this.filesDatabase.data));
```

Comment:
Does the `data` field include the workspace root? We'll need both the path to the file and the workspace root to determine the relative path that needs to be created on the local disk.

Reply:
I think there's no need to have the workspace in the `data` because it's already in the info that I'm getting from the selected files to download in the `file-explorer.component.ts`

Reply:
Maybe there's a problem with the way state is being stored right now. Right now `TransferablesGridComponent` has a single object `FilesDatabase` which contains `destination` and `preserveStructure`. But there could be different downloads running at the same time that have different settings for these objects. So this state should be stored in the model that backs `TransferableItemComponent` instead.

I think we need a model that contains items that are downloading (or are queued/paused in a download). Each downloading object contains its source path and its destination path; the original values for `destination` and `preserveStructure` would no longer be necessary at this point. It looks like `FilesDatabase` is this model, but the model doesn't have all the data we need yet.

It's important that the transferable grid be able to contain more than one set of selected downloads at a time, each of which could have a different set of download settings.

It also occurs to me that each downloadable item may have its own download controller, because in the case where we allow upload, download and copy between different services, we'd need to make the pause/resume/abort operations behave differently based on the item type. This controller (or service) would also be set at the time the downloading object is first created.